src/gc.c

```diff
@@ -104,8 +99,12 @@ void *GC_thread(void *val)
 						// Forwarded to an upstream DNS server
 						// Adjust counters
 						counters->forwarded--;
-						if(upstream != NULL)
-							upstream->count--;
+						if(query->upstreamID > -1)
+						{
+							upstreamsData* upstream = getUpstream(query->upstreamID, true);
```

Comment:
This is the actual fix: Do not try to access the data of an upstream destination with ID < 0 (this is causing the wrong warning here).